src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/AWSSecretsManagerPlugin.java

```diff
@@ -179,7 +179,19 @@ public void openInitialConnection(ConnectionUrl connectionUrl) throws SQLExcepti
    */
   private boolean isLoginUnsuccessful(SQLException exception) {
     this.logger.logTrace("Login failed. SQLState=" + exception.getSQLState(), exception);
-    return SQLSTATE_ACCESS_ERROR.equals(exception.getSQLState());
+
+    Throwable throwable = exception;
+    while (throwable != null) {
+      if (throwable instanceof SQLException) {
+        if (SQLSTATE_ACCESS_ERROR.equals(((SQLException) throwable).getSQLState())) {
+          return true;
+        }
+
+        throwable = throwable.getCause();
```

Comment:
I think it makes sense to move this line outside of if(). We want to iterate through all nested exceptions, not just through SQLExceptions. Like this:

while(throwable != null) {
if (throwale instanceof SQLException) {
...}
throwable = throwable.getCause();
}